en/08_Changelogs/6.0.0.md

```diff
@@ -1043,14 +1045,28 @@ As part of this change, the [`FormField::validate()`](api:SilverStripe\Forms\For
 
 This is no longer supported. Instead, you should override the [`DataObject::getCMSCompositeValidator()`](api:SilverStripe\ORM\DataObject::getCMSCompositeValidator()) method as described in [validation in the CMS](/developer_guides/forms/validation/#validation-in-the-cms).
 
+### `CMSMain` search filter changes {#cmsmain-search-filter}
+
+The search filter in the pages admin section (aka [`CMSMain`](api:SilverStripe\CMS\Controllers\CMSMain)) is visually similar to the search filters in other sections of the CMS, but has until now had a completely different code path and therefore slightly different functionality.
+
+In CMS 6 that functionality now follows the same code paths used by [`GridFieldFilterHeader`](api:SilverStripe\Forms\GridFieldGridFieldFilterHeader) which powers the search filters in most other sections of the CMS.
+
+This means `CMSMain` search filters now rely on `SearchContext` to perform the actual filtering, and get their configuration from the [`searchable_fields`](api:SilverStripe\ORM\DataObject->searchable_fields) configuration property. Since `CMSMain` uses [`SiteTree`](api:SilverStripe\CMS\Model\SiteTree) as its model class by default, you'll need to update the `searchable_fields` configuration for `SiteTree` via YAML configuration if you want to update the fields users can filter against.
+
+Shared functionality for building the search filter form has been moved from `CMSMain` and `GridFieldFilterHeader` into a new [`SearchContextForm`](api:SilverStripe\ORM\Search\SearchContextForm) class. Some of the HTML markup for the search filters has also been updated so it can be shared across all CMS sections.
+
+One consequence of this is that [`CMSSiteTreeFilter`](api:SilverStripe\CMS\Controllers\CMSSiteTreeFilter) and its subclasses are no long primarily responsible for filtering in the pages admin section. They are still used for filtering by the status (e.g. draft vs modified) of pages. If you had implemented a custom `CMSSiteTreeFilter` subclass, you will need to either modify it (if it is intended to provide functionality for filtering by a status) or look into refactoring it into a replacement or extension for the [`SiteTreeSearchContext`](api:SilverStripe\CMS\Search\SiteTreeSearchContext) class.
+
+This also means that the [`ElementSiteTreeFilterSearch`](api:DNADesign\Elemental\Controllers\ElementSiteTreeFilterSearch) class and its functionality are no longer available. This means when searching for pages, the content of elemental blocks will not be included in the search. Given that functionality relied on [`DataList::filterByCallback()`](api:SilverStripe\ORM\DataList::filterByCallback()) (and therefore looped through *every single page* and could cause performance problems) we suspect its usage was low. If you need that functionality, implementing a custom `SiteTreeSearchContext` class would probably be the way to go.
```

Comment:
```suggestion
This also means that the [`ElementSiteTreeFilterSearch`](api:DNADesign\Elemental\Controllers\ElementSiteTreeFilterSearch) class and its functionality are no longer available. This means that when searching for pages, the content of elemental blocks will not be included in the search. Given that functionality relied on [`DataList::filterByCallback()`](api:SilverStripe\ORM\DataList::filterByCallback()) (and therefore looped through *every single page* and could cause performance problems) we suspect its usage was low. If you need that functionality, implementing a custom `SiteTreeSearchContext` class would probably be the way to go.
```

Reply:
I think it works fine either way but done to avoid unnecessary ping pong